tensorrt_llm/_torch/modules/linear.py

```diff
@@ -369,7 +369,11 @@ def apply_linear(self,
                 raise ValueError(f'unsupported quant mode: {qc.quant_mode}')
         else:
             # TODO: remove custom cublas_mm when default heuristics is good enough
-            if self.use_custom_cublas_mm:
+            if input.shape[0] >= 1 and input.shape[0] <= 16 and input.shape[
+                    1] == 7168 and weight.shape[0] == 2112:
+                output = torch.ops.trtllm.fuse_a_gemm_op(
+                    input, self.weight.t(), bias, None)
```

Comment:
This should be not be added to linear module. @QiJune to advice based on llama4 approach.